de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/environment/AddedFilesEnvironmentManager.java

```diff
@@ -115,5 +111,10 @@ public void deleteEnvironment(Path path) {
     clearAddedFiles();
     manager.deleteEnvironment(path);
   }
+  
+  @Override
+  public String[] getEntries() {
+    return entries;
```

Comment:
@ahmadswaid: this method returns only the added files. The other environmentManager classes return all files associated with getEnvironment(). Fixing this isn't easy. If you change the method to return added files AND the files from the "sub-manager", then you need to change the handling of added files in FSKEditorJSNodeDialog as well. I was unable to do that, somehow on clicking "ok" in the dialog, all the entries were duplicated. I pushed my changes to a separate branch, maybe you can figure out what to do here. [see here](https://github.com/SiLeBAT/FSK-Lab/tree/addedFiles_problem)


1st Editor: add directory
![image](https://user-images.githubusercontent.com/38949263/198299666-617b53e8-39f9-4d64-bdda-4cb257977160.png)

2nd Editor, Add File "hello.py"
![image](https://user-images.githubusercontent.com/38949263/198299902-3a908213-8ff0-4d8d-afe7-b42644f1d8f8.png)

On reopening the Dialog:
![image](https://user-images.githubusercontent.com/38949263/198300150-58cddff5-d02a-47e7-9893-a4b515cce1c6.png)

Reply:
Thanks @schuelet, I noticed this, I'm working on it, because of that I didn't ask for review :)
Actually it seems to be an old problem.

Reply:
@schuelet this should be ready, please have the review. thxs

Reply:
@ahmadswaid the idea of the AddedFilesManager was to allow to add files to an existing model. Miguels Environment implementation didn't allow for that use case, so I tried to build an environment that carries its "parent environment" and in addition tracks the files that were added by the "add files" button. 
As of now, the files added by using this "add file" button overwrite the existing files, that's what I meant in an earlier comment ("getEntries "returns only the added files", not all files from existing environments").
<img width="869" alt="image" src="https://user-images.githubusercontent.com/38949263/198546020-29e35aa4-4929-4be1-a815-47e941d662d3.png">

<img width="339" alt="image" src="https://user-images.githubusercontent.com/38949263/198546156-4943849a-8382-47b1-8d20-668236bfecc9.png">


 I was unable to fix it (see images and branch above), and unfortunately the issue remains in the latest commit.